code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,42 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("failed to generate Deepmaint - no Deepmaint Z-levels registered!")
+		return
+
+	for(var/area/map_template/deepmaint_wfc/deepmaint_area in GLOB.wfc_deepmaint_zlevels_by_instance)
+		var/success = deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
+		if(success)
+			log_and_message_admins("has generated a DeepmaintRooms map.")
+			break
+		else
+			log_and_message_admins("has unsuccessfully attempted to generate a DeepmaintRooms map.")
+
+	return
+
```

Comment:
Redundant
```suggestion
```

Reply:
Yes, but I hate implicit returns; you have to think more to guess the return value and it makes refactoring uglier if a diff accidentally joins two procs together.

Reply:
Eh. I personally stick to whatever code style the rest of the code uses, which in this case is implicit returns if you're not returning a var directly. I'm not fussed too much either way, but it's not really done anywhere else so stands out.

I believe this has been addressed in 515 though. iirc you can, much in the style of python, define a return type for a proc.